client/src/main.rs

```diff
@@ -111,24 +113,66 @@ fn main() {
         Commands::StartTally { election_id: _ } => {
             println!("Starting tally ...");
         }
-        Commands::StoreMessage { message_data } => {
+        Commands::VoteFor { message_data } => {
             println!("Generating the transaction...");
             println!("This may take a while");
 
             //TO DO: Let the user make an account and use it, instead
             //of creating it with a random one
+
             let transaction =
                 transactions::create_store_data_transaction(message_data.as_bytes().to_vec(), true);
-
             let encoded_data = hex::encode(transaction.to_bytes_le().unwrap());
             println!("The transaction hexdata is: \n {}", encoded_data);
             println!("Sending transactions to multiple nodes ...");
             // To improve reliability we send the transaction to many nodes
-            let results = rcp::sync_spray_transaction(encoded_data);
-            println!("The nodes responses are: ");
-            for result in results {
-                println!("{}", result);
+            let responses = rcp::sync_spray_transaction(encoded_data);
+
+            let mut iter = responses.iter();
+            let mut ok_response: Value = json!("");
+            let mut amount_of_bad_results = 0;
+            for response in responses.iter() {
+                match response {
+                    Ok(value) => {
+                        ok_response = value.clone();
+                        break;
+                    }
+                    Err(_) => {
+                        amount_of_bad_results = amount_of_bad_results + 1;
+                    }
+                }
+            }
+
+            if responses.len() == amount_of_bad_results {
+                println!("Aleo nodes can't be reached, try again later");
+                return;
+            }
+
+            println!("Nodes received the transaction succesfully");
+
+            let transaction_id = ok_response.get("result").unwrap();
+
+            println!("The transaction id is: {}", transaction_id);
+
+            println!("Notifying the transaction submission to the tallying server ...");
+
+            let request_json = json!({ "aleo_transaction_id": transaction_id });
+
+            let client = reqwest::blocking::Client::new();
+            let send_result = client
+                .post("http://127.0.0.1:3000/election/msg")
+                .json(&request_json)
+                .send();
+
+            match send_result {
+                Ok(v) => {
```

Comment:
It would be nice to match a 201 code (CREATED), not just an OK.

https://docs.rs/reqwest/0.9.20/reqwest/struct.Response.html